### DISCUSSION TITLE
Problemi con firmware di ITEAD Sonoff ZigBee 3.0 USB Dongle Plus

### DISCUSSION THREAD

**Discussion by Dim-1** [indomus/forum#1714] [Hardware & Firmware]
Seguendo la procedura per aggiornare la Sonoff ZigBee 3.0 USB Dongle Plus presente sul sito di indomus, mi si blocca al lancio della riprogrammazione

```
(base) [dim@fedora:~/cc2538-bsl]$ sudo ./cc2538-bsl.py -p /dev/ttyUSB0 -evw --bootloader-sonoff-usb CC1352P2_CC2652P_launchpad_coordinator_20220219.hex
./cc2538-bsl.py requires the Python serial library
Please install it with:

   pip3 install pyserial
```
pyserial  è installato (ho seguito la guida), infatti
```
(base) [dim@fedora:~/cc2538-bsl]$ pip3 install pyserial
Requirement already satisfied: pyserial in /home/dim/anaconda3/lib/python3.8/site-packages (3.5)
```
Questo avviene sia dal portatile (da cui ho tratto il codice postato sopra) che sul minipc:
```
deb@debian:~/cc2538-bsl$ sudo ./cc2538-bsl.py -p /dev/ttyUSB0 -evw --bootloader-sonoff-usb CC1352P2_CC2652P_launchpad_coordinator_20220219.hex
./cc2538-bsl.py requires the Python serial library
Please install it with:

   pip3 install pyserial
deb@debian:~/cc2538-bsl$   pip3 install pyserial
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: pyserial in /home/deb/.local/lib/python3.10/site-packages (3.5)
```
Suggerimenti? Grazie

**Comment by snakuzzo** [comment]
Se vedi bene...stai eseguendo lo script con `sudo` e quindi come utente `root`...mentre poi vai ad installare senza `sudo` con utente `deb`.
Quindi installi su un environment e poi lo esegui da un altro...ed è normale che non funzioni.

**Reply by Dim-1** [reply]
Hai ragione e installando come utente ```deb```, ha funzionato. Ho usato ```sudo``` perché lo script non partiva per problemi di permessi. Ho risolto cambiando i permessi a tutti i file coinvolti nello script, ```sudo chmod -R 777 <nomecartella>```, e poi sono andato avanti come utente normale.
Grazie

**Comment by Dim-1** [comment]
Grazie per la risposta,
Senza il sudo non partiva lo script, mi pare di ricordare, non sono davanti
il PC adesso.  Riproverò da capo con scrupolo, anche se mi sembrava di aver
fatto bene, sudo a parte(e comunque l'ho inserito dopo aver provato il
comando senza essere il superuser). Grazie

Il mer 1 giu 2022, 11:00 Diego Ambrosanio ***@***.***> ha
scritto:
